yarn-project/builder/src/contract-interface-gen/typescript.ts

```diff
@@ -45,9 +45,18 @@ function abiTypeToTypescript(type: ABIParameter['type']): string {
         return 'WrappedFieldLike';
       }
       if (isBoundedVecStruct(type)) {
-        // To make BoundedVec easier to work with, we expect a simple array on the input and then we encode it
-        // as a BoundedVec in the ArgumentsEncoder.
-        return `${abiTypeToTypescript(type.fields[0].type)}`;
+        // This generates a readonly array of the max length of the bounded vec.
+        // E.g. for BoundedVec<Field, 3> we would get:
+        // "readonly [FieldLike?,FieldLike?,FieldLike?]"
```

Comment:
By having each of the items in the readonly array be optional we make the type compatible with arrays smaller than max length. So for `readonly [FieldLike?,FieldLike?,FieldLike?]` passing there an array defined as `const array = [1n, 2n] as const` would work.